apis/duck/unstructured_test.go

```diff
@@ -70,23 +73,87 @@ func TestFromUnstructuredFooable(t *testing.T) {
 		want:      FooStatus{},
 		wantError: nil,
 	}}
+
 	for _, tc := range tcs {
-		raw, err := json.Marshal(tc.in)
-		if err != nil {
-			panic("failed to marshal")
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			raw, err := json.Marshal(tc.in)
+			if err != nil {
+				panic("failed to marshal")
+			}
+
+			t.Logf("Marshalled : %s", string(raw))
+
+			got := Foo{}
+			err = FromUnstructured(tc.in, &got)
+			if err != tc.wantError {
+				t.Fatalf("Unexpected error for %q: %v", string(tc.name), err)
+			}
+
+			if !reflect.DeepEqual(tc.want, got.Status) {
+				t.Errorf("Decode(%q) want: %+v\ngot: %+v", string(tc.name), tc.want, got)
+			}
+		})
+	}
+}
 
-		t.Logf("Marshalled : %s", string(raw))
+func TestToUnstructured(t *testing.T) {
+	tests := []struct {
+		name      string
+		in        OneOfOurs
+		want      *unstructured.Unstructured
+		wantError error
+	}{{
+		name: "missing TypeMeta",
+		in: &Resource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "blah",
+			},
+		},
+		want: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "pkg.knative.dev/v2",
+				"kind":       "Resource",
+				"metadata": map[string]interface{}{
+					"creationTimestamp": nil,
+					"name":              "blah",
+				},
+				"spec": map[string]interface{}{},
+			},
+		},
+	}, {
+		name: "with TypeMeta",
+		in: &Resource{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "blah",
+			},
+		},
+		want: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "apps/v1",
+				"kind":       "Deployment",
+				"metadata": map[string]interface{}{
+					"creationTimestamp": nil,
+					"name":              "blah",
+				},
+				"spec": map[string]interface{}{},
+			},
+		},
+	}}
 
-		got := Foo{}
-		err = FromUnstructured(tc.in, &got)
-		if err != tc.wantError {
-			t.Errorf("Unexpected error for %q: %v", string(tc.name), err)
-			continue
-		}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := ToUnstructured(tc.in)
+			if err != tc.wantError {
+				t.Fatalf("Unexpected error for %q: %v", string(tc.name), err)
```

Comment:
Test name is already present in `t.Run` call, so no need to print it again.

Reply:
Done